pkg/cluster/kubernetes/namespacer.go

```diff
@@ -51,7 +47,7 @@ var getDefaultNamespace = func() (string, error) {
 		return c.Namespace, nil
 	}
 
-	return defaultFallbackNamespace, nil
+	return defaultNamespace, nil
```

Comment:
I think that the passed namespace should take precedence over the kubeconfig file mounted. So, I would:

1. Rename this function to `getKubeconfigDefaultNamespace()` and leave `defaultFallbackNamespace` as it was
2. Rename the `defaultNamespace` parameter of `NewNamespacer` to `defaultNamespaceOverride` and set it to the `fallback` without calling `getKubeconfigDefaultNamespace()` if `!= ""`